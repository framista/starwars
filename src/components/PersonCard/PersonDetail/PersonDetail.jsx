import React from 'react';
import PropTypes from 'prop-types';

const PersonDetail = ({ detail }) => {
  const [label, value] = detail;
  return (
    <div className="capitalize">
      {label} - {value}
    </div>
  );
};

PersonDetail.propTypes = {
  detail: PropTypes.arrayOf(PropTypes.string),
};

export default React.memo(PersonDetail);
