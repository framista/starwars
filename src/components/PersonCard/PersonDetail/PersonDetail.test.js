import { render, screen } from '@testing-library/react';

import PersonDetail from './PersonDetail';

const detail = ['name', 'Luke Skywalker'];

describe('PersonDetail', () => {
  it('should contain suitable text', () => {
    render(<PersonDetail detail={detail} />);
    expect(screen.getByText('name - Luke Skywalker')).toBeInTheDocument();
  });
  it('should render correctly', () => {
    const { asFragment } = render(<PersonDetail detail={detail} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
