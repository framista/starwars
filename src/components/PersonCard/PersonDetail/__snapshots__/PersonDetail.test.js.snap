// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetail should render correctly 1`] = `
<DocumentFragment>
  <div
    class="capitalize"
  >
    name - Luke Skywalker
  </div>
</DocumentFragment>
`;
